GAME/TRAPS.C

```diff
@@ -124,8 +128,15 @@ void LavaBurn(struct ITEM_INFO* item)//5AD78, 5B1F4
 
 void LaraBurn()//5ACE4, 5B160
 {
-	S_Warn("[LaraBurn] - Unimplemented!\n");
-	return;
+	if (!lara.burn && !lara.BurnSmoke)
+	{
+		short fx = CreateEffect(lara_item->room_number);
```

Comment:
Variables must be declared at the very beginning of a method.

Reply:
C89 states that variables must be declared at the start of a scope block, which is the case here